zetkin/app.zetkin.org
# issue-424/add jest to project

Resolves #424

Changes:
* Adds jest and `@testing-library/react` to project and configure
  * Create custom `render` function which implements required providers for testing components
* Update README to include Jest testing instructions
* Test for `CampaignActionButtons` to test public page button navigates to correct path (validate react implementation of jest works)
* Test for `stringUtils`, `stringToBool` (validate non-react implementation of jest)
* Fixes small ts errors in `TimeFrame` that arose due to installing `ts-jest`
* Update CI
  * Use `actions/setup-node@v2` instead of `v1`, which supports yarn dependency caching
  * Lint action no longer checks types
  * New action to check types
  * New action to run jest unit tests
